src/libraries/System.Linq/tests/AnyTests.cs

```diff
@@ -64,6 +96,26 @@ public void Any(IEnumerable<int> source, bool expected)
             Assert.Equal(expected, source.Any());
         }
 
+        [Theory]
+        [MemberData(nameof(TestDataWithPredicate))]
+        public void AnyList(IEnumerable<int> source, Func<int, bool> predicate, bool expected) {
+            if (predicate == null) {
+                Assert.Equal(expected, new List<int>(source).Any());
+            } else {
+                Assert.Equal(expected, new List<int>(source).Any(predicate));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TestDataWithPredicate))]
+        public void AnyArray(IEnumerable<int> source, Func<int, bool> predicate, bool expected) {
+            if (predicate == null) {
+                Assert.Equal(expected, System.Linq.Enumerable.ToArray(source).Any());
+            } else {
+                Assert.Equal(expected, System.Linq.Enumerable.ToArray(source).Any(predicate));
+            }
+        }
+
```

Comment:
Nit: I think we've normally been putting tests with `MemberData` _below_ the data.  Certainly new tests should likely be placed with other tests of its type.

Otherwise, see earlier comments about `All_TestData`.